File: js/source.js
function minifyLua(code) {
    let strippedCode = code.replace(/\s+/g, " ");
    let minifiedCode = strippedCode.replace(/\s*([\(\)\{\}\[\],;])\s*/g, "$1");
    minifiedCode = minifiedCode.replace(/;(\s*\})/g, "$1");
    return minifiedCode;
}

function unminifyLua(code) {
    let formattedCode = code.replace(/;/g, ";\n");
    formattedCode = formattedCode.replace(/([^\w])function/g, "$1\nfunction");
    formattedCode = formattedCode.replace(/end([^\w])/g, "end\n$1");
    formattedCode = formattedCode.replace(/([=%+\-*/%])/g, " $1 ");
    return formattedCode;
}

File: rust/source.rs
fn minify_lua(code: &str) -> String {
    let stripped_code = code.replace(char::is_whitespace, " ");
    let minified_code = stripped_code.replace(r"\s*([\(\)\{\}\[\],;])\s*", "$1");
    let minified_code = minified_code.replace(r";(\s*\})", "$1");
    minified_code
}

fn unminify_lua(code: &str) -> String {
    let formatted_code = code.replace(";", ";\n");
    let formatted_code = formatted_code.replace(r"([^\w])function", "$1\nfunction");
    let formatted_code = formatted_code.replace(r"end([^\w])", "end\n$1");
    let formatted_code = formatted_code.replace(r"([=%+\-*/%])", " $1 ");
    formatted_code
}